src/main/java/bio/terra/service/iam/LocalAuthenticatedUserRequestFactory.java

```diff
@@ -28,15 +32,20 @@ public LocalAuthenticatedUserRequestFactory(ApplicationConfiguration application
   public AuthenticatedUserRequest from(HttpServletRequest servletRequest) {
     HttpServletRequest req = servletRequest;
 
-    Optional<String> token =
+    String token =
         Optional.ofNullable(req.getHeader("Authorization"))
-            .map(header -> header.substring("Bearer ".length()));
+            .map(header -> header.substring("Bearer ".length()))
+            .orElse("");
 
     String email =
         Optional.ofNullable(req.getHeader("From")).orElse(applicationConfiguration.getUserEmail());
 
     String userId = applicationConfiguration.getUserId();
 
-    return new AuthenticatedUserRequest().email(email).subjectId(userId).token(token);
+    return AuthenticatedUserRequest.builder()
+        .setEmail(email)
+        .setSubjectId(userId)
+        .setToken(token)
+        .build();
```

Comment:
Do you know what happens if we hit the "orElse" condition and the `token = ""`?

Reply:
Looking through some of the other comments - maybe we need to just throw an error if it's null?  But, then, like Muscles mentioned I'm not sure what would happen with our unauthenticated endpoints.

Reply:
If the token is null, the AuthenticatedUserRequest would throw an error: https://github.com/DataBiosphere/terra-common-lib/blob/0.0.51/src/main/java/bio/terra/common/iam/AuthenticatedUserRequest.java#L80

Reply:
If the token is null, the AuthenticatedUserRequest would throw an error: https://github.com/DataBiosphere/terra-common-lib/blob/0.0.51/src/main/java/bio/terra/common/iam/AuthenticatedUserRequest.java#L80

Reply:
We hit the `orElse` condition when you're not authenticated, and so if the endpoint requires authentication you'll get a 401 error and unauthenticated endpoints still work (just tested this out locally).